### PR TITLE
fix(photon): allow reconnecting after disconnect

### LINKED ISSUE #273: Allow reconnecting after disconnect
As of now, I get the error `Engine has stopped` and can't reconnect a Photon instance.

### PR DESCRIPTION
**Pull request by timsuchanek**
Closes #273

### CODE REVIEW

File: packages/photon/fixtures/blog/main.ts [hunk 1]
```diff
@@ -1,18 +1,39 @@
...
-    errorFormat: 'pretty',
+  const photon = new Photon()
+
+  const user1 = await photon.users.create({
+    data: {
+      email:
+        'a@a.de2' +
+        Math.random()
+          .toString(36)
+          .substring(2, 15),
```
Comment (Jolg42) [inline packages/photon/fixtures/blog/main.ts]: Weird emails :)

File: packages/photon/src/__tests__/runtime-tests/blog/index.js [hunk 2]
```diff
@@ -5,4 +5,11 @@ module.exports = async () => {
     errorFormat: 'colorless',
   })
   await photon.users()
+  photon.disconnect()
+  await photon.users()
+  photon.disconnect()
+  photon.connect()
+  photon.disconnect()
+  photon.connect()
```
Comment (Jolg42) [inline packages/photon/src/__tests__/runtime-tests/blog/index.js]: Should we add a setTimeout to reconnect after x seconds maybe?

File: packages/photon/src/generation/TSClient.ts [hunk 3]
```diff
@@ -432,30 +433,36 @@ export class Photon {
...
       return this.connectionPromise
     }
-    this.connectionPromise = this.connectEngine(true)
+    this.connectionPromise = this.engine.start()
     return this.connectionPromise!
   }
   async disconnect() {
-    await this.engine.stop()
+    if (this.connectionPromise) {
+      await this.connectionPromise
```
Comment (Jolg42) [inline packages/photon/src/generation/TSClient.ts]: Is there a timeout on the connectionPromise?

Comment (timsuchanek) [reply packages/photon/src/generation/TSClient.ts]: Yes, in `NodeEngine`!

Comment (timsuchanek) [reply packages/photon/src/generation/TSClient.ts]: Yes, in `NodeEngine`!

**Pull request merged**